canvas_sdk/data/task.py

```diff
@@ -0,0 +1,61 @@
+from datetime import datetime
+from enum import Enum
+
+from pydantic_partial import create_partial_model
+
+from canvas_sdk.data import PluginModel
+from canvas_sdk.data.patient import Patient
+from canvas_sdk.data.staff import Staff
+from canvas_sdk.data.team import Team
+from canvas_sdk.effects import Effect, EffectType
+
+
+class Task(PluginModel):
+    class Status(Enum):
+        COMPLETED = "Completed"
+        CLOSED = "Closed"
+        OPEN = "Open"
+
+    id: str | None = None
+    creator: Staff
+    assignee: Staff | None = None
+    patient: Patient | None = None
+    team: Team | None = None
+    title: str
+    due: datetime | None = None
+    status: Status = Status.OPEN
+    comments: "list[TaskComment]" = []
+    labels: list[str] = []
+
+    def create(self) -> Effect:
+        return Effect(type=EffectType.CREATE_TASK, payload=self.as_json())
+
+    def update(self) -> Effect:
+        if not self.id:
+            raise ValueError("Cannot update a Task without an id")
+        payload = self.as_json(exclude_unset=True)
+        return Effect(type=EffectType.UPDATE_TASK, payload=payload)
+
+    def add_comment(self, comment: str) -> Effect:
+        if not self.id:
+            raise ValueError("Cannot add a comment to a Task without an id")
+        task_comment = TaskComment(task=self, body=comment)
+        return Effect(
+            type=EffectType.CREATE_TASK_COMMENT, payload=task_comment.as_json(exclude_unset=True)
+        )
+
+    # TODO - add:
+    # add task comment
+
+
+class TaskComment(PluginModel):
+    id: str | None = None
+    task: Task | None = None
+    body: str
+
+
+Task.model_rebuild()
```

Comment:
Just curious, what does this do?

Reply:
It makes it so `TaskComment` can be referenced in the `Task` model as a string, even though `TaskComment` is defined later in the file. i.e.:

```
comments: "list[TaskComment] | None" = []
```

Reply:
Information about this:

https://docs.pydantic.dev/latest/concepts/models/#rebuild-model-schema